src/node/rpc/test/frontend_test.cpp

```diff
@@ -451,3 +467,115 @@ int main(int argc, char** argv)
     return res;
   return res;
 }
+
+class StubForwarder : public AbstractForwarder
+{
+public:
+  std::vector<std::vector<uint8_t>> forwarded_cmds;
+
+  StubForwarder() {}
+
+  bool forward_command(
+    enclave::RpcContext& rpc_ctx,
+    NodeId from,
+    NodeId to,
+    CallerId caller_id,
+    const std::vector<uint8_t>& data) override
+  {
+    forwarded_cmds.push_back(data);
+    return true;
+  }
+
+  void clear()
+  {
+    forwarded_cmds.clear();
+  }
+};
+
+TEST_CASE("Forwarding")
+{
+  prepare_callers();
+
+  TestForwardingFrontEnd frontend_follower(*network.tables);
+  TestForwardingFrontEnd frontend_leader(*network2.tables);
+
+  // Set forwarder and replicator for follower frontend
```

Comment:
It's preferable to have comments state why things are done a certain why, rather than re-state what the code does, unless it is particularly counter-intuitive. Here, the names are sensible, and these are plain assignments, so I think the comment isn't helpful (same on line 507).